File: AdvantageFramework/Database/Procedures/ClientPortalAlertRecipient.vb
Namespace Database.Procedures.ClientPortalAlertRecipient

    <HideModuleName()> _
    Public Module Methods

#Region " Constants "



#End Region

#Region " Enum "



#End Region

#Region " Variables "



#End Region

#Region " Properties "



#End Region

#Region " Methods "

        Public Function LoadByAlertID(ByVal DbContext As AdvantageFramework.Database.DbContext, ByVal AlertID As Long) As System.Data.Entity.Infrastructure.DbQuery(Of AdvantageFramework.Database.Entities.ClientPortalAlertRecipient)

            LoadByAlertID = From ClientPortalAlertRecipient In DbContext.GetQuery(Of Database.Entities.ClientPortalAlertRecipient)
                            Where ClientPortalAlertRecipient.AlertID = AlertID
                            Select ClientPortalAlertRecipient

        End Function
        Public Function LoadByAlertIDAndContactID(ByVal DbContext As AdvantageFramework.Database.DbContext, ByVal AlertID As Long, ByVal ClientContactID As Integer) As AdvantageFramework.Database.Entities.ClientPortalAlertRecipient

            LoadByAlertIDAndContactID = (From ClientPortalAlertRecipient In DbContext.GetQuery(Of Database.Entities.ClientPortalAlertRecipient)
                                         Where ClientPortalAlertRecipient.AlertID = AlertID _
                                        And ClientPortalAlertRecipient.ClientContactID = ClientContactID
                                         Select ClientPortalAlertRecipient).SingleOrDefault

        End Function

        Public Function Load(ByVal DbContext As AdvantageFramework.Database.DbContext) As System.Data.Entity.Infrastructure.DbQuery(Of AdvantageFramework.Database.Entities.ClientPortalAlertRecipient)

            Load = From ClientPortalAlertRecipient In DbContext.GetQuery(Of Database.Entities.ClientPortalAlertRecipient)
                   Select ClientPortalAlertRecipient

        End Function
        Public Function Insert(ByVal DbContext As AdvantageFramework.Database.DbContext, ByVal AlertID As Integer, _
                               ByVal ClientContactID As Integer, ByVal EmailAddress As String, ByVal ProcessedDate As Nullable(Of Date), _
                               ByVal IsNewAlert As Nullable(Of Short), ByVal HasBeenRead As Nullable(Of Short), _
                               ByRef ClientPortalAlertRecipient As AdvantageFramework.Database.Entities.ClientPortalAlertRecipient) As Boolean

            'objects
            Dim Inserted As Boolean = False

            Try

                ClientPortalAlertRecipient = New AdvantageFramework.Database.Entities.ClientPortalAlertRecipient

                ClientPortalAlertRecipient.DbContext = DbContext
                ClientPortalAlertRecipient.AlertID = AlertID
                ClientPortalAlertRecipient.ClientContactID = ClientContactID
                ClientPortalAlertRecipient.EmailAddress = EmailAddress
                ClientPortalAlertRecipient.ProcessedDate = ProcessedDate
                ClientPortalAlertRecipient.IsNewAlert = IsNewAlert
                ClientPortalAlertRecipient.HasBeenRead = HasBeenRead

                Inserted = Insert(DbContext, ClientPortalAlertRecipient)

            Catch ex As Exception
                Inserted = False
            Finally
                Insert = Inserted
            End Try

        End Function
        Public Function Insert(ByVal DbContext As AdvantageFramework.Database.DbContext, ByVal ClientPortalAlertRecipient As AdvantageFramework.Database.Entities.ClientPortalAlertRecipient) As Boolean

            'objects
            Dim Inserted As Boolean = False
            Dim IsValid As Boolean = True
            Dim ErrorText As String = ""

            Try

                DbContext.ClientPortalAlertRecipients.Add(ClientPortalAlertRecipient)

                ErrorText = ClientPortalAlertRecipient.ValidateEntity(IsValid)

                If IsValid Then

                    If ClientPortalAlertRecipient.AlertRecipientID = 0 Then

                        Try

                            ClientPortalAlertRecipient.AlertRecipientID = (From Entity In AdvantageFramework.Database.Procedures.ClientPortalAlertRecipient.LoadByAlertID(DbContext, ClientPortalAlertRecipient.AlertID) _
                                                                           Select Entity.AlertRecipientID).Max + 1

                        Catch ex As Exception
                            ClientPortalAlertRecipient.AlertRecipientID = 1
                        End Try

                    End If

                    DbContext.SaveChanges()

                    Inserted = True

                Else

                    AdvantageFramework.Navigation.ShowMessageBox(ErrorText)

                End If

            Catch ex As Exception
                Inserted = False
            Finally
                Insert = Inserted
            End Try

        End Function
        Public Function Update(ByVal DbContext As AdvantageFramework.Database.DbContext, ByVal ClientPortalAlertRecipient As AdvantageFramework.Database.Entities.ClientPortalAlertRecipient) As Boolean

            'objects
            Dim Updated As Boolean = False
            Dim IsValid As Boolean = True
            Dim ErrorText As String = ""

            Try

                DbContext.UpdateObject(ClientPortalAlertRecipient)

                ErrorText = ClientPortalAlertRecipient.ValidateEntity(IsValid)

                If IsValid Then

                    DbContext.SaveChanges()

                    Updated = True

                Else

                    AdvantageFramework.Navigation.ShowMessageBox(ErrorText)

                End If

            Catch ex As Exception
                Updated = False
            Finally
                Update = Updated
            End Try

        End Function
        Public Function Delete(ByVal DbContext As AdvantageFramework.Database.DbContext, ByVal ClientPortalAlertRecipient As AdvantageFramework.Database.Entities.ClientPortalAlertRecipient) As Boolean

            'objects
            Dim Deleted As Boolean = False
            Dim IsValid As Boolean = True
            Dim ErrorText As String = ""

            Try

                If IsValid Then
                    
                    DbContext.DeleteEntityObject(ClientPortalAlertRecipient)

                    DbContext.SaveChanges()

                    Deleted = True

                Else

                    AdvantageFramework.Navigation.ShowMessageBox(ErrorText)

                End If

            Catch ex As Exception
                Deleted = False
            Finally
                Delete = Deleted
            End Try

        End Function

#End Region

    End Module

End Namespace

File: AdvantageFramework.Core/Database/Procedures/ClientPortalAlertRecipient.cs
using System;
using System.Linq;

namespace AdvantageFramework.Core.Database.Procedures
{
    public static partial class ClientPortalAlertRecipient
    {

        #region  Constants 



        #endregion

        #region  Enum 



        #endregion

        #region  Variables 



        #endregion

        #region  Properties 



        #endregion

        #region  Methods 

        public static System.Linq.IQueryable<Entities.ClientPortalAlertRecipient> LoadByAlertID(AdvantageFramework.Core.Database.DbContext DbContext, long AlertID)
        {
            System.Linq.IQueryable<Entities.ClientPortalAlertRecipient> LoadByAlertIDRet = default;
            LoadByAlertIDRet = from ClientPortalAlertRecipient in DbContext.ClientPortalAlertRecipients.AsQueryable()
                               where ClientPortalAlertRecipient.AlertId == AlertID
                               select ClientPortalAlertRecipient;
            return LoadByAlertIDRet;
        }

        public static AdvantageFramework.Core.Database.Entities.ClientPortalAlertRecipient LoadByAlertIDAndContactID(AdvantageFramework.Core.Database.DbContext DbContext, long AlertID, int ClientContactID)
        {
            AdvantageFramework.Core.Database.Entities.ClientPortalAlertRecipient LoadByAlertIDAndContactIDRet = default;
            LoadByAlertIDAndContactIDRet = (from ClientPortalAlertRecipient in DbContext.ClientPortalAlertRecipients.AsQueryable()
                                            where ClientPortalAlertRecipient.AlertId == AlertID & ClientPortalAlertRecipient.CdpContactId == ClientContactID
                                            select ClientPortalAlertRecipient).SingleOrDefault();
            return LoadByAlertIDAndContactIDRet;
        }

        public static System.Linq.IQueryable<Entities.ClientPortalAlertRecipient> Load(AdvantageFramework.Core.Database.DbContext DbContext)
        {
            System.Linq.IQueryable<Entities.ClientPortalAlertRecipient> LoadRet = default;
            LoadRet = from ClientPortalAlertRecipient in DbContext.ClientPortalAlertRecipients.AsQueryable()
                      select ClientPortalAlertRecipient;
            return LoadRet;
        }

        //public static bool Insert(AdvantageFramework.Core.Database.DbContext DbContext, int AlertID, int ClientContactID, string EmailAddress, DateTime? ProcessedDate, short? IsNewAlert, short? HasBeenRead, ref AdvantageFramework.Core.Database.Entities.ClientPortalAlertRecipient ClientPortalAlertRecipient)


        //{
        //    bool InsertRet = default;

        //    // objects
        //    bool Inserted = false;
        //    try
        //    {
        //        ClientPortalAlertRecipient = new AdvantageFramework.Core.Database.Entities.ClientPortalAlertRecipient();
        //        ClientPortalAlertRecipient.DbContext = DbContext;
        //        ClientPortalAlertRecipient.AlertID = AlertID;
        //        ClientPortalAlertRecipient.ClientContactID = ClientContactID;
        //        ClientPortalAlertRecipient.EmailAddress = EmailAddress;
        //        ClientPortalAlertRecipient.ProcessedDate = ProcessedDate;
        //        ClientPortalAlertRecipient.IsNewAlert = IsNewAlert;
        //        ClientPortalAlertRecipient.HasBeenRead = HasBeenRead;
        //        Inserted = Insert(DbContext, ClientPortalAlertRecipient);
        //    }
        //    catch (Exception ex)
        //    {
        //        Inserted = false;
        //    }
        //    finally
        //    {
        //        InsertRet = Inserted;
        //    }

        //    return InsertRet;
        //}

        //public static bool Insert(AdvantageFramework.Core.Database.DbContext DbContext, AdvantageFramework.Core.Database.Entities.ClientPortalAlertRecipient ClientPortalAlertRecipient)
        //{
        //    bool InsertRet = default;

        //    // objects
        //    bool Inserted = false;
        //    bool IsValid = true;
        //    string ErrorText = "";
        //    try
        //    {
        //        DbContext.ClientPortalAlertRecipients.Add(ClientPortalAlertRecipient);
        //        ErrorText = ClientPortalAlertRecipient.ValidateEntity(IsValid);
        //        if (IsValid)
        //        {
        //            if (ClientPortalAlertRecipient.AlertRecipientID == 0)
        //            {
        //                try
        //                {
        //                    ClientPortalAlertRecipient.AlertRecipientID = (from Entity in AdvantageFramework.Core.Database.Procedures.ClientPortalAlertRecipient.LoadByAlertID(DbContext, ClientPortalAlertRecipient.AlertID)
        //                                                                   select Entity.AlertRecipientID).Max + 1;
        //                }
        //                catch (Exception ex)
        //                {
        //                    ClientPortalAlertRecipient.AlertRecipientID = 1;
        //                }
        //            }

        //            DbContext.SaveChanges();
        //            Inserted = true;
        //        }
        //        else
        //        {
        //            //AdvantageFramework.Navigation.ShowMessageBox(ErrorText);
        //        }
        //    }
        //    catch (Exception)
        //    {
        //        Inserted = false;
        //    }
        //    finally
        //    {
        //        InsertRet = Inserted;
        //    }

        //    return InsertRet;
        //}

        //public static bool Update(AdvantageFramework.Core.Database.DbContext DbContext, AdvantageFramework.Core.Database.Entities.ClientPortalAlertRecipient ClientPortalAlertRecipient)
        //{
        //    bool UpdateRet = default;

        //    // objects
        //    bool Updated = false;
        //    bool IsValid = true;
        //    string ErrorText = "";
        //    try
        //    {
        //        DbContext.UpdateObject(ClientPortalAlertRecipient);
        //        ErrorText = ClientPortalAlertRecipient.ValidateEntity(IsValid);
        //        if (IsValid)
        //        {
        //            DbContext.SaveChanges();
        //            Updated = true;
        //        }
        //        else
        //        {
        //            AdvantageFramework.Navigation.ShowMessageBox(ErrorText);
        //        }
        //    }
        //    catch (Exception ex)
        //    {
        //        Updated = false;
        //    }
        //    finally
        //    {
        //        UpdateRet = Updated;
        //    }

        //    return UpdateRet;
        //}

        //public static bool Delete(AdvantageFramework.Core.Database.DbContext DbContext, AdvantageFramework.Core.Database.Entities.ClientPortalAlertRecipient ClientPortalAlertRecipient)
        //{
        //    bool DeleteRet = default;

        //    // objects
        //    bool Deleted = false;
        //    bool IsValid = true;
        //    string ErrorText = "";
        //    try
        //    {
        //        if (IsValid)
        //        {
        //            DbContext.DeleteEntityObject(ClientPortalAlertRecipient);
        //            DbContext.SaveChanges();
        //            Deleted = true;
        //        }
        //        else
        //        {
        //            //AdvantageFramework.Navigation.ShowMessageBox(ErrorText);
        //        }
        //    }
        //    catch (Exception)
        //    {
        //        Deleted = false;
        //    }
        //    finally
        //    {
        //        DeleteRet = Deleted;
        //    }

        //    return DeleteRet;
        //}

        #endregion

    }
}
